src/utils/filter-utils.js

```diff
@@ -304,12 +368,74 @@ export function getFieldDomain(data, field) {
   }
 }
 
+export const getPolygonFilterFunctor = (layer, filter) => {
+  const getPosition = layer.getPositionAccessor();
+
+  switch (layer.type) {
+    case LAYER_TYPES.arc:
+    case LAYER_TYPES.line:
+      return data => {
+        const pos = getPosition({data});
+        return pos.every(Number.isFinite) && [
+          [pos[0], pos[1]],
+          [pos[3], pos[4]]
+        ].every(point => isInPolygon(point, filter.value));
+      };
+    default:
+      return data => {
+        const pos = getPosition({data});
+        return pos.every(Number.isFinite) && isInPolygon(pos, filter.value);
+      }
+  }
+};
+
+/**
+ * @param field dataset Field
+ * @param dataId Dataset id
+ * @param filter Filter object
+ * @param layers list of layers to filter upon
+ * @return {*}
+ */
+export function getFilterFunction(field, dataId, filter, layers) {
+  // field could be null
+  const valueAccessor = data => field ? data[field.tableFieldIndex - 1] : null;
+
+  switch (filter.type) {
+    case FILTER_TYPES.range:
+      return data => isInRange(valueAccessor(data), filter.value);
+    case FILTER_TYPES.multiSelect:
+      return data => filter.value.includes(valueAccessor(data));
+    case FILTER_TYPES.select:
+      return data => valueAccessor(data) === filter.value;
+    case FILTER_TYPES.timeRange:
+      const mapppedValue = get(field, ['filterProp', 'mappedValue']);
+      const accessor = Array.isArray(mapppedValue) ?
+        (data, index) => field.filterProp.mappedValue[index]
```

Comment:
You already have `mapppedValue` defined
`(data, index) => mappedValue[index] : data => timeToUnixMilli(valueAccessor(data), field.format);`